math/src/fft/fft_inputs.rs

```diff
@@ -172,3 +335,65 @@ pub(super) fn fft_in_place<B, I>(
         }
     }
 }
+
+// TRANSPOSING
+// ================================================================================================
+
+#[cfg(feature = "concurrent")]
+fn transpose_square_stretch<E, I>(matrix: &mut I, size: usize, stretch: usize)
+where
+    E: FieldElement,
+    I: FftInputs<E> + ?Sized,
+{
+    assert_eq!(matrix.len(), size * size * stretch);
```

Comment:
Shouldn't this be `debug_assert_eq`? If we really need to assert that here, then we probably should add a comment on why this is a production panic, and extend this comment to the public API that invokes this function.